exercises/concept/dancing-dots/test/dancing_dots/animation_test.exs

```diff
@@ -0,0 +1,199 @@
+      assert DancingDots.Flicker.handle_frame(dot2, 17, []) == %{dot2 | opacity: 0.6}
+      assert DancingDots.Flicker.handle_frame(dot2, 18, []) == %{dot2 | opacity: 0.6}
+      assert DancingDots.Flicker.handle_frame(dot2, 19, []) == %{dot2 | opacity: 0.6}
+      assert DancingDots.Flicker.handle_frame(dot2, 20, []) == %{dot2 | opacity: 0.3}
+    end
+
+    @tag task_id: 3
+    test "can be used in a dot group" do
+      dot1 = %DancingDots.Dot{x: 44, y: 44, radius: 2, opacity: 0.8}
+      dot2 = %DancingDots.Dot{x: 0, y: 0, radius: 3, opacity: 0.5}
+
+      {:ok, dot_group} = DancingDots.DotGroup.new([dot1, dot2], [{DancingDots.Flicker, []}])
+
+      assert dot_group ==
+               %DancingDots.DotGroup{
+                 animations_with_opts: [{DancingDots.Flicker, []}],
+                 dots: [dot1, dot2]
+               }
+
+      assert DancingDots.DotGroup.render_dots(dot_group, 4) == [
+               %{dot1 | opacity: 0.4},
+               %{dot2 | opacity: 0.25}
+             ]
+    end
+  end
+
+  describe "Zoom module" do
+    @tag task_id: 4
+    test "implements Animation behaviour" do
+      Code.ensure_loaded(DancingDots.Zoom)
+      assert function_exported?(DancingDots.Zoom, :init, 1)
+      assert function_exported?(DancingDots.Zoom, :handle_frame, 3)
+    end
+
+    @tag task_id: 4
+    test "has a custom init/1 implementation that checks that a valid velocity was passed as an option" do
+      assert DancingDots.Zoom.init(velocity: 10) == {:ok, [velocity: 10]}
+
+      assert DancingDots.Zoom.init([]) ==
+               {:error, "Expected required option :velocity to be a number, got: nil"}
+
+      assert DancingDots.Zoom.init(velocity: "7") ==
+               {:error, "Expected required option :velocity to be a number, got: \"7\""}
+    end
+
+    @tag task_id: 4
+    test "the first frame, handle_frame/3 returns the dot unchanged" do
+      dot = %DancingDots.Dot{x: 90, y: 90, radius: 100, opacity: 0.5}
+
+      assert DancingDots.Zoom.handle_frame(dot, 1, velocity: 3) == dot
+      assert DancingDots.Zoom.handle_frame(dot, 1, velocity: 30) == dot
+      assert DancingDots.Zoom.handle_frame(dot, 1, velocity: -7) == dot
+    end
+
+    @tag task_id: 4
+    test "every frame after the first, handle_frame/3 grows the dot's radius by the given velocity" do
+      dot1 = %DancingDots.Dot{x: 90, y: 90, radius: 100, opacity: 0.5}
+      dot2 = %DancingDots.Dot{x: 100, y: 0, radius: 400, opacity: 0.9}
+
+      assert DancingDots.Zoom.handle_frame(dot1, 1, velocity: 3) == %{dot1 | radius: 100}
+      assert DancingDots.Zoom.handle_frame(dot1, 2, velocity: 3) == %{dot1 | radius: 103}
+      assert DancingDots.Zoom.handle_frame(dot1, 3, velocity: 3) == %{dot1 | radius: 106}
+      assert DancingDots.Zoom.handle_frame(dot1, 101, velocity: 3) == %{dot1 | radius: 400}
+
+      assert DancingDots.Zoom.handle_frame(dot2, 1, velocity: -1) == %{dot2 | radius: 400}
```

Comment:
How about splitting this test in two here and mention negative velocity? Just because the instructions mention it but the test descriptions don't.